tests/integration/offer/test_fixed_unit_benefit.py

```diff
@@ -0,0 +1,368 @@
+        condition_range.add_product(self.condition_product)
+        self.condition = models.CountCondition.objects.create(
+            range=condition_range, type=models.Condition.COUNT, value=2
+        )
+
+        self.benefit_product = factories.ProductFactory()
+        benefit_range = factories.RangeFactory()
+        benefit_range.add_product(self.benefit_product)
+        self.benefit = models.FixedUnitDiscountBenefit.objects.create(
+            range=benefit_range, type=models.Benefit.FIXED_UNIT, value=D("3.00")
+        )
+
+        self.offer = models.ConditionalOffer(
+            id=1, condition=self.condition, benefit=self.benefit
+        )
+        self.basket = factories.create_basket(empty=True)
+
+        self.applicator = Applicator()
+
+    def test_succcessful_application_consumes_correctly(self):
+        add_product(self.basket, product=self.condition_product, quantity=2)
+        add_product(self.basket, product=self.benefit_product, quantity=1)
+
+        self.applicator.apply_offers(self.basket, [self.offer])
+
+        discounts = self.basket.offer_applications.offer_discounts
+        self.assertEqual(len(discounts), 1)
+        self.assertEqual(discounts[0]["freq"], 1)
+
+    def test_condition_is_consumed_correctly(self):
+        # Testing an error case reported on the mailing list
+        add_product(self.basket, product=self.condition_product, quantity=3)
+        add_product(self.basket, product=self.benefit_product, quantity=2)
+
+        self.applicator.apply_offers(self.basket, [self.offer])
+
+        discounts = self.basket.offer_applications.offer_discounts
+        self.assertEqual(len(discounts), 1)
+        self.assertEqual(discounts[0]["freq"], 1)
+
+
+class TestAbsoluteProductDiscountAppliedWithCountCondition(TestCase):
+    def setUp(self):
+        product_range = models.Range.objects.create(
+            name="All products", includes_all_products=True
+        )
+        self.condition = models.CountCondition.objects.create(
+            range=product_range, type=models.Condition.COUNT, value=2
+        )
+        self.offer = mock.Mock()
+        self.benefit = models.FixedUnitDiscountBenefit.objects.create(
+            range=product_range, type=models.Benefit.FIXED_UNIT, value=D("3.00")
+        )
+        self.basket = factories.create_basket(empty=True)
+
+    def test_applies_correctly_to_empty_basket(self):
+        result = self.benefit.apply(self.basket, self.condition, self.offer)
+        self.assertEqual(D("0.00"), result.discount)
+        self.assertEqual(0, self.basket.num_items_with_discount)
+        self.assertEqual(0, self.basket.num_items_without_discount)
+
+    def test_applies_correctly_to_basket_which_matches_condition_with_one_line(self):
+        add_product(self.basket, price=D("12.00"), quantity=2)
+        result = self.benefit.apply(self.basket, self.condition, self.offer)
+        self.assertEqual(D("3.00"), result.discount)
```

Comment:
I still believe this should be 6, 3 dollar discount per product, because there are 2 products and each gets 3 dollar discount.